oci/table_oci_file_storage_snapshot.go

```diff
@@ -0,0 +1,262 @@
+)
+
+//// TABLE DEFINITION
+
+func tableFileStorageSnapshot(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "oci_file_storage_snapshot",
+		Description: "OCI File Storage Snapshot",
+		Get: &plugin.GetConfig{
+			KeyColumns:        plugin.SingleColumn("id"),
+			ShouldIgnoreError: isNotFoundError([]string{"400"}),
+			Hydrate:           getFileStorageSnapshot,
+		},
+		List: &plugin.ListConfig{
+			Hydrate:       listFileStorageSnapshot,
+			ParentHydrate: listFileStorageFileSystems,
+		},
+		GetMatrixItem: BuildCompartementZonalList,
+		Columns: []*plugin.Column{
+			{
+				Name:        "name",
+				Description: "Name of the snapshot.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "id",
+				Description: "The OCID of the snapshot.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromCamel(),
+			},
+			{
+				Name:        "lifecycle_state",
+				Description: "The current state of the snapshot.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "time_created",
+				Description: "The date and time the snapshot was created.",
+				Type:        proto.ColumnType_TIMESTAMP,
+				Transform:   transform.FromField("TimeCreated.Time"),
+			},
+			{
+				Name:        "file_system_id",
+				Description: "The OCID of the file system from which the snapshot was created.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("FileSystemId"),
+			},
+			{
+				Name:        "provenance_id",
+				Description: "An OCID identifying the parent from which this snapshot was cloned.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("ProvenanceId"),
+			},
+			{
+				Name:        "is_clone_source",
+				Description: "Specifies whether the snapshot has been cloned.",
+				Type:        proto.ColumnType_BOOL,
+			},
+			{
+				Name:        "lifecycle_details",
+				Description: "Additional information about the current 'lifecycleState'.",
+				Type:        proto.ColumnType_STRING,
+			},
+
+			// tags
+			{
+				Name:        "defined_tags",
+				Description: ColumnDescriptionDefinedTags,
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "freeform_tags",
+				Description: ColumnDescriptionFreefromTags,
+				Type:        proto.ColumnType_JSON,
+			},
+
+			//  Steampipe standard columns
+			{
+				Name:        "tags",
+				Description: ColumnDescriptionTags,
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.From(snapshotTags),
+			},
+			{
+				Name:        "title",
+				Description: ColumnDescriptionTitle,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Name"),
+			},
+
+			// Standard OCI columns
+			{
+				Name:        "region",
+				Description: ColumnDescriptionRegion,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Id").Transform(ociRegionName),
+			},
+			{
```

Comment:
Add `compartment_id` column also.

Reply:
If we are not getting compartment_id in `file_storage_snapshot` then we can fetch it from parent `file_storage_file_system`.